app/src/main/java/org/commcare/commcaresupportlibrary/identity/IdentityResponseBuilder.java

```diff
@@ -0,0 +1,104 @@
+     */
+    public static IdentityResponseBuilder registrationResponse(String guid) {
+        Intent intent = new Intent();
+        intent.putExtra(REGISTRATION, new RegistrationResult(guid));
+        return new IdentityResponseBuilder(intent);
+    }
+
+    /**
+     * Creates response for result of a identification workflow
+     *
+     * @param identificationMatches list of matches to be passed back to CommCare
+     * @return IdentityResponseBuilder for a identification workflow response
+     */
+    public static IdentityResponseBuilder identificationResponse(ArrayList<IdentificationMatch> identificationMatches) {
+        Intent intent = new Intent();
+        intent.putParcelableArrayListExtra(IDENTIFICATION, identificationMatches);
+        return new IdentityResponseBuilder(intent);
+    }
+
+    /**
+     * Creates response for result of a verification workflow
+     *
+     * @param guid        Identity Provider's GUID for the beneficiary the verification was performed
+     * @param matchResult Result of the verification process
+     * @return IdentityResponseBuilder for a verification workflow response
+     */
+    public static IdentityResponseBuilder verificationResponse(String guid, MatchResult matchResult) {
+        Intent intent = new Intent();
+        intent.putExtra(VERIFICATION, new VerificationMatch(guid, matchResult));
+        return new IdentityResponseBuilder(intent);
+    }
+
+    /**
+     * Creates response with a list of duplicates identified in the registration workflow
+     *
+     * @param identificationMatches
+     * @return IdentityResponseBuilder for a registration workflow response
+     */
+    public static IdentityResponseBuilder registrationResponse(ArrayList<IdentificationMatch> identificationMatches) {
+        Intent intent = new Intent();
+        intent.putExtra(REGISTRATION_DUPLICATES, true);
+        intent.putParcelableArrayListExtra(IDENTIFICATION, identificationMatches);
+        return new IdentityResponseBuilder(intent);
+    }
+
+    /**
+     * Adds identifications to a response. This can be used in a registration workflow to pass back possible duplicates
+     * for a new registration
+     *
+     * @param identificationMatches list of matches against a biometric
+     * @return IdentityResponseBuilder containing the list of {@code identificationMatches}
+     */
+    public IdentityResponseBuilder setIdentificationMatches(ArrayList<IdentificationMatch> identificationMatches) {
+        resultIntent.putParcelableArrayListExtra(IDENTIFICATION, identificationMatches);
+        return this;
+    }
+
+    // Can be used to get the response Intent from the IdentityResponseBuilder
+    public Intent build() {
+        return resultIntent;
+    }
+
+    // Finalize the response and return result back to CommCare
+    public void finalizeResponse(Activity activity) {
+        activity.setResult(RESULT_OK, resultIntent);
```

Comment:
Might be good to run an integrity check here (or above) which identifies things like whether this has all of the pieces it needs for a response to be valid.

Reply:
hmm, I think I am going to mark the constructor for `IdentityResponseBuilder` as `private` here instead so that all instances of `IdentityResponseBuilder` happen using one of the factory menthods which make sure that `IdentityResponseBuilder` is build correctly.